rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -335,7 +372,16 @@ void Player::prepare_clock(rcutils_time_point_value_t starting_time)
 {
   // TODO(emersonknapp) move clock setup into the constructor
   double rate = play_options_.rate > 0.0 ? play_options_.rate : 1.0;
+  bool paused = false;
+  // Copy over important settings
+  if (clock_) {
+    rate = clock_->get_rate();
+    paused = clock_->is_paused();
+  }
   clock_ = std::make_unique<rosbag2_cpp::TimeControllerClock>(starting_time, rate);
+  if (paused) {
+    clock_->pause();
+  }
```

Comment:
@emersonknapp It seems that we need to add `paused` inside constructor of `rosbag2_cpp::TimeControllerClock` as a parameter or initialize it by default to true. Otherwise if we had been in pause and doing clock re-init here, the clock might advance since creation when we will hit pause here.

Reply:
Good comment - this is handled by https://github.com/ros2/rosbag2/pull/754 since we no longer re-create the clock.

Reply:
Yeahh.. I just reviewed it. I agree that it will not cause problem when it will be in constructor.